test/e2e/antreapolicy_test.go

```diff
@@ -75,13 +75,13 @@ type TestCase struct {
 // TestStep is a single unit of testing spec. It includes the policy specs that need to be
 // applied for this test, the port to test traffic on and the expected Reachability matrix.
 type TestStep struct {
-	Name         string
-	Reachability *Reachability
-	Policies     []metav1.Object
-	Groups       []metav1.Object
-	Port         []int32
-	Duration     time.Duration
-	CustomProbes []*CustomProbe
+	Name              string
+	Reachability      *Reachability
+	Policies          []metav1.Object
+	ServicesAndGroups []metav1.Object
```

Comment:
I was thinking that by bundling the two, we can easily control and test different scenarios such as Service created first vs. Group created first. This can simply be controlled by the relative position of the obj in the slice.

Reply:
By this reasoning, shouldn't policies be included the the same slice as Services and Groups as well? I don't feel strongly about it however, so the current code is fine.

Reply:
That's a good point.... I think I'll change it to having all resources in a single slice.

Reply:
For CGs we impose the restriction that a CG must exist before referenced in ACNP.... so CGs should always be created before policies and cleaned up after policies.

Reply:
> For CGs we impose the restriction that a CG must exist before referenced in ACNP.... so CGs should always be created before policies and cleaned up after policies.

I remember that we have this. BTW I wonder if this was the right choice. It doesn't match how other object references are validated. For example, the ServiceReference in CG can refer to a Service that does not exist yet (in which case the CG is empty). But anyway, this is orthogonal to this PR. The current solution works for me.